sql/src/main/java/io/druid/sql/calcite/rel/DruidQueryBuilder.java

```diff
@@ -368,18 +369,27 @@ public TopNQuery toTopNQuery(
       final boolean useApproximateTopN
   )
   {
-    // Must have GROUP BY one column, ORDER BY one column, limit less than maxTopNLimit, and no HAVING.
-    if (grouping == null
-        || grouping.getDimensions().size() != 1
-        || limitSpec == null
-        || limitSpec.getColumns().size() != 1
-        || limitSpec.getLimit() > maxTopNLimit
-        || having != null) {
+    // Must have GROUP BY one column, ORDER BY zero or one column, limit less than maxTopNLimit, and no HAVING.
```

Comment:
A few lines later, if order/limit is on the metric and `useApproximateTopN` is false, then we return null meaning topN is not usable.